primes.rb

```diff
@@ -1,6 +1,29 @@
 class Primes
+
   def self.sum_to(limit = 100)
-    # TODO - add your prime number solution here...
-    "I'm working on it!"
+
+  	#Using sieve of eratosthenes to find sum of primes to limit
+
+  	#Set up array from 0 to limit
+    primes = (0..limit).to_a
+    #Set 0 and 1 to nil because they are not primes
+    primes[0] = primes[1] = nil
+
+    #Go through prime candidates
+    primes.each do |p|
+      #Go past if nil
+      next unless p
+      #Break if past square root of limit
+      break if p*p > limit
+      #Start at square of current, step through, go up to limit by multiples of current number, make each nil
+      (p*p).step(limit,p) { |m| primes[m] = nil }
+    end
+
+    #Remove nils
+    sum = primes.compact
+    #Sum of each prime in array
+    sum.inject(:+)
```

Comment:
This code looks great from a coding perspective and is easy enough to follow. Be aware though that your line 8 is going to be 'expensive' in terms of how much memory and space that you'll need- imagine an array with 2 million entries in it! We'll be able to teach you some other approaches that would make this more efficient - well done.